cdhist/cdhist.py

```diff
@@ -11,16 +11,42 @@
 
 # Following is template for the shell code injected into your session
 SHELLCODE = '''
-!cmd() {
+%cmd() {
     local d
-    d=$(!prog "$@")
+    # shellcheck disable=SC2164
+    d="$(%prog "$@")" && builtin cd -- "$d"
+}
 
-    if [ $? -ne 0 ]; then
-        return 0
-    fi
+# ======================================================================
+# Hook to add new entries to the history
 
-    builtin cd -- "$d"
-}
+if [[ -n $BASH_VERSION ]]; then
+    __cdhist_oldpwd="$(pwd)"
+
+    function __cdhist_prompt_hook() {
+        local -r retval pwd_tmp
+        retval="$?"
+        pwd_tmp="$(pwd)"
+
+        if [[ $__cdhist_oldpwd != "$pwd_tmp" ]]; then
```

Comment:
why this oldpwd trickery? why can't this be simple as zsh part?

Reply:
Bash does not provide an entry point for hooks that shall only be executed on cwd changes. Hence, ones needs to utilise the `PROMPT_COMMAND` entry point.

Reply:
actually, there is a library you could use:
- https://github.com/mpapis/bash_zsh_support

Reply:
Hm, interesting. I am not sure whether it is worth the effort though since this script needs to be installed as well.

Reply:
yes, adding it's support here only adds complexity. it's more informative post.